portal/patient/libs/Controller/PatientController.php

```diff
@@ -188,14 +185,13 @@ public function Create()
     {
         try {
             $json = json_decode(RequestUtil::GetBody());
-            if (! $json) {
+            if (empty($json)) {
                 throw new Exception('The request body does not contain valid JSON');
             }
-
+            if ($_SESSION['pid'] !== true && $_SESSION['register'] !== true) {
+                throw new Exception('Unauthorized');
+            }
             $patient = new Patient($this->Phreezer);
-// this is an auto-increment. uncomment if updating is allowed
-            // $patient->Id = $this->SafeGetVal($json, 'id');
-
             $patient->Title = $this->SafeGetVal($json, 'title', $patient->Title);
```

Comment:
change line 275 to:
```
$patient->Save(true);
```
(just to double ensure only an insert will take place with this call)